pkg/agent/route/route_linux.go

```diff
@@ -1873,27 +1926,39 @@ func (c *Client) DeleteEgressRoutes(tableID uint32) error {
 	return nil
 }
 
-func (c *Client) AddEgressRule(tableID uint32, mark uint32) error {
+func (c *Client) AddEgressRule(tableID uint32, mark uint32, isIPv6 bool) error {
+	family := netlink.FAMILY_V4
+	if isIPv6 {
+		family = netlink.FAMILY_V6
+	}
 	rule := netlink.NewRule()
 	rule.Table = int(tableID)
 	rule.Mark = mark
 	rule.Mask = ptr.To(types.SNATIPMarkMask)
+	rule.Family = family
```

Comment:
Is Egress broken on IPv6-only clusters currently?

In any case, could you have that change in a separate PR?